src/components/Search/SearchRouter/SearchRouter.tsx

```diff
@@ -1,27 +1,57 @@
+import {useNavigationState} from '@react-navigation/native';
 import debounce from 'lodash/debounce';
-import React, {useCallback, useState} from 'react';
+import React, {useCallback, useMemo, useState} from 'react';
 import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
 import FocusTrapForModal from '@components/FocusTrap/FocusTrapForModal';
 import Modal from '@components/Modal';
+import {useOptionsList} from '@components/OptionListContextProvider';
 import type {SearchQueryJSON} from '@components/Search/types';
 import useKeyboardShortcut from '@hooks/useKeyboardShortcut';
 import useResponsiveLayout from '@hooks/useResponsiveLayout';
 import useThemeStyles from '@hooks/useThemeStyles';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
 import * as SearchUtils from '@libs/SearchUtils';
 import Navigation from '@navigation/Navigation';
+import variables from '@styles/variables';
 import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
 import ROUTES from '@src/ROUTES';
 import {useSearchRouterContext} from './SearchRouterContext';
 import SearchRouterInput from './SearchRouterInput';
+import SearchRouterList from './SearchRouterList';
 
 const SEARCH_DEBOUNCE_DELAY = 200;
 
 function SearchRouter() {
     const styles = useThemeStyles();
-
+    const [betas] = useOnyx(`${ONYXKEYS.BETAS}`);
     const {isSmallScreenWidth} = useResponsiveLayout();
     const {isSearchRouterDisplayed, closeSearchRouter} = useSearchRouterContext();
+
     const [currentQuery, setCurrentQuery] = useState<SearchQueryJSON | undefined>(undefined);
+    const contextualReportID = useNavigationState<Record<string, {reportID: string}>, string | undefined>((state) => {
+        return state.routes.at(-1)?.params?.reportID;
+    });
+    const [recentSearches] = useOnyx(ONYXKEYS.RECENT_SEARCHES);
+    const sortedRecentSearches = Object.values(recentSearches ?? {}).sort((a, b) => {
+        const dateA = new Date(a.timestamp);
+        const dateB = new Date(b.timestamp);
+        return dateB.getTime() - dateA.getTime();
+    });
```

Comment:
I think we could do `useMemo` for sorting - in this case it will never be heavy, but its cleaner/safer to add it